src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -135,16 +134,23 @@ public static void finalizeRegion(final AssemblyRegion region,
         if ( region.isFinalized() ) {
             return;
         }
-
         final byte minTailQualityToUse = errorCorrectReads ? HaplotypeCallerEngine.MIN_TAIL_QUALITY_WITH_ERROR_CORRECTION : minTailQuality;
 
         final List<GATKRead> readsToUse = new ArrayList<>();
         final List<GATKRead> hardClippedReadsToUse = new ArrayList<>();
 
         for (final GATKRead originalRead : region.getReads()) {
+
+            boolean isFlowBasedData = false;
+            if (FlowBasedReadUtils.hasFlowTags(originalRead)) {
```

Comment:
Can this be condensed into a single method so external tools don't have to know about the readgroup info caching? Can you make a method `FlowBasedReadUtils.isThisFlowBased()` that extracts this entire operation away and handles your readgroup caching internally?